### ISSUE TITLE
Comments on posts

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#4926]
Looks like members of the public can't add updates/comments to published posts. I think this would be a valuable feature. (via Nekesa Were)

Aha! Link: [https://ushahiditeam.aha.io/features/PROD-747](https://ushahiditeam.aha.io/features/PROD-747)

[View original issue in GitHub](https://github.com/ushahidi/platform/issues/296)

---

**rjmackay** 2015-09-02

cc @jshorland
Any other requirements for comments from BRCK?

**jshorland** 2015-10-27

Nice to have feature (feedback from quakemap who found comments very valuable): getting notifications when someone comments on a post you've also commented on. Similar to blog commenting notifications.

**rjmackay** 2016-04-05

@jshorland bumping this to the next sprint

**snilsmitchell** 2016-04-27

Is this now in Sprint #2 (May 2 - June 3)?

**rjmackay** 2016-04-29

@sebm90 its probably going to get bumped further out. Is there a solutions project waiting on that??

**jshorland** 2016-05-02

#508 is related

**jshorland** 2017-08-29

Closing #508 in favor of this and adding comment here:

Copying V2 feature request 595: Switching "on" comments for logged in users

I don't see a way to limit commenting to logged in users only - only the on/off switch in the site settings. Would be great to add this for both spam protection and accountability. Best would be to assign it as a user role capability that can be selected for assorted user levels.

Nice as an option, but don't want to limit comments to logged-in users on every Ushahidi deployment. Ushahidi would be a rather limited crowd-sourcing tool if the crowd can't comment. BTW, Akismet has done a great job of eliminating spam for us.

**jshorland** 2017-08-29

This might align nicely with future of "tasks"

**rowasc** 2019-06-07

[https://github.com/ushahidi/platform-client/pull/1229](https://github.com/ushahidi/platform-client/pull/1229)
[https://github.com/ushahidi/platform/pull/3207](https://github.com/ushahidi/platform/pull/3207)
Noting this 2 PRs from Will that seem to be addressing some of the "comments" functionality, will need to review and test/demo to see where we are exactly on this

**Erioldoesdesign** 2019-07-10

I would also like to do user testing on this when it goes out. I suspect we should make this 'optional' via admin settings to avoid 'trolling' if that isn't already a specification.

**rowasc** 2019-07-10

Agree on making it optional. 
I'll let you know when this lands in a test environment and you can run user tests there?

**Erioldoesdesign** 2019-07-10

That'd be awesome! Usertesting in the test environment is great cause we often gather other insights about other features 🙌

**rowasc** 2019-09-03

Ok.  :( Sadly the PR still needs work (probably because it's also super stale  😬 ) . It's not running, and the migrations do not run, so we will

* close the PR so it's not there just sitting in our queue without doing anything.
* keep the branches so we can come back to it.
* keep this issue open and link to the branches..
  * [https://github.com/ushahidi/platform-client/pull/1229](https://github.com/ushahidi/platform-client/pull/1229) client work
  * [https://github.com/ushahidi/platform/pull/3207](https://github.com/ushahidi/platform/pull/3207) backend work
